nomad/resource_acl_auth_method.go

```diff
@@ -0,0 +1,407 @@
+					api.ACLAuthMethodTokenLocalityLocal,
+				}, false),
+			},
+			"max_token_ttl": {
+				Description: "Defines the maximum life of a token created by this method.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"default": {
+				Description: "Defines whether this ACL Auth Method is to be set as default.",
+				Optional:    true,
+				Type:        schema.TypeBool,
+			},
+			"config": {
+				Description: "",
+				Required:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Elem:        resourceACLAuthMethodConfig(),
+			},
+		},
+	}
+}
+
+func resourceACLAuthMethodConfig() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"oidc_discovery_url": {
+				Description: "",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"oidc_client_id": {
+				Description: "",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"oidc_client_secret": {
+				Description: "",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+			},
+			"oidc_scopes": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"bound_audiences": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"allowed_redirect_uris": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    true,
+			},
+			"discovery_ca_pem": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"signing_algs": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"claim_mappings": {
+				Description: "",
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"list_claim_mappings": {
+				Description: "",
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceACLAuthMethodCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	aclAuthMethod, err := generateNomadACLAuthMethod(d)
+	if err != nil {
+		return err
+	}
+
+	// Create our ACL auth method.
+	log.Print("[DEBUG] Creating ACL Auth Method")
+	aclBindingRuleCreateResp, _, err := client.ACLAuthMethods().Create(aclAuthMethod, nil)
+	if err != nil {
+		return fmt.Errorf("error creating ACL Auth Method: %s", err.Error())
+	}
+	log.Printf("[DEBUG] Created ACL Auth Method %q", aclBindingRuleCreateResp.Name)
+	d.SetId(aclBindingRuleCreateResp.Name)
```

Comment:
Just double checking: is the name guaranteed to be unique?

Reply:
Yes.